x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/core/tree/Node.java

```diff
@@ -367,31 +368,36 @@ public List<Object> nodeProperties() {
     }
 
     public String nodeString() {
+        return nodeString(propertiesToString(true /* skipIfChild */));
+    }
+
+    /**
+     * Like the above, but ensures consistent tree rendering for golden tests. Overrides of this method should avoid adding information
+     * that can change between repeated test runs, such as object identities or pseudo-random generated names.
+     */
+    public String goldenTestNodeString() {
+        return nodeString(goldenTestPropertiesToString(true /* skipIfChild */));
+    }
+
+    private String nodeString(String propertiesToString) {
         StringBuilder sb = new StringBuilder();
         sb.append(nodeName());
         sb.append("[");
-        sb.append(propertiesToString(true));
+        sb.append(propertiesToString);
         sb.append("]");
         return sb.toString();
     }
 
     @Override
     public String toString() {
-        return treeString(new StringBuilder(), 0, new BitSet()).toString();
+        return treeString(new StringBuilder(), 0, new BitSet(), Node::nodeString).toString();
     }
 
-    /**
-     * Render this {@link Node} as a tree like
-     * <pre>
-     * {@code
-     * Project[[i{f}#0]]
-     * \_Filter[i{f}#1]
-     *   \_SubQueryAlias[test]
-     *     \_EsRelation[test][i{f}#2]
-     * }
-     * </pre>
-     */
-    final StringBuilder treeString(StringBuilder sb, int depth, BitSet hasParentPerDepth) {
+    public String goldenTestToString() {
```

Comment:
nit: It would be good to have some javadocs here, explaining the difference between `goldenTestToString` and `goldenTestNodeString`